### PR TITLE
Coordinator with live data from participants

### PR DESCRIPTION
**Pull request by campbellsean**
Also a switch away from React Table

### CODE REVIEW

File: src/components/Navigation.js [hunk 1]
```diff
@@ -23,23 +22,43 @@ import Progress from "./Progress"
 const Navigation = observer(({ store }) => (
   <Layout>
     <InternalLink to={Home} store={store} >
-      <Icon path={mdiHome} color={store.currentPage == Home ? primary : black} size="1.8rem" />
+      <Icon
```
Comment (campbellsean) [inline src/components/Navigation.js]: Should this be a rule of thumb when you have 3 attributes?

File: src/primitives/TreatmentProgress.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import React from "react"
+import styled from "styled-components"
+import { observer } from "mobx-react"
+
+import CircularProgressbar from "react-circular-progressbar"
+import { primary } from "../colors"
+
```
Comment (campbellsean) [inline src/primitives/TreatmentProgress.js]: I see, this is CustomProgressBar but with some refactoring? Looks a lot better

File: src/Assembly.js [hunk 3]
```diff
@@ -203,24 +194,38 @@ class Assembly extends React.Component {
...
+  // ------ Misc ------
+
+  @observable language = "Español"
+  @observable alerts = []
+  @observable currentPage = Login
+
   constructor(props) {
     super(props)
 
-    if(!this.authorized) { this.currentPage = Login }
```
Comment (campbellsean) [inline src/Assembly.js]: We will need this to keep not authorized users outside of our app

File: src/components/Navigation.js (hunk 1 above)
Comment (thecalliopecrashed) [reply src/components/Navigation.js]: Yeah, I try to break up long lines as much as possible.

Any line over 80 characters is too long, according to many devs.

More practically, anything that you have to scroll sideways to see could be better presented on multiple lines.

File: src/primitives/TreatmentProgress.js (hunk 2 above)
Comment (thecalliopecrashed) [reply src/primitives/TreatmentProgress.js]: Yep! Renamed the file, moved some styles into a `styled-component`.

File: src/Assembly.js (hunk 3 above)
Comment (thecalliopecrashed) [reply src/Assembly.js]: Don't worry!
On line 183, we set the default page to `Login`.

You're right, though - we can be doing more to keep un-authenticated people out.
This line on its own just redirects the user when they first hit the app; it doesn't force people to log in.

I created #95 to follow up on this.

File: models/participant.rb [hunk 4]
```diff
@@ -10,4 +10,31 @@ class Participant < ApplicationRecord
...
+      notes: notes,
+
+      today: {
+        medication_reports: medication_reports.where("timestamp >= :date", date: Time.current.beginning_of_day),
+        strip_reports: strip_reports.where("timestamp >= :date", date: Time.current.beginning_of_day),
+        symptom_reports: symptom_reports.where("created_at >= :date", date: Time.current.beginning_of_day),
+        notes: notes.where("created_at >= :date", date: Time.current.beginning_of_day),
+      }
+    }
+  end
```
Comment (thecalliopecrashed) [inline models/participant.rb]: This is the new data structure that shows up on the client for participants.

File: src/components/AdherenceCalendar.js [hunk 5]
```diff
@@ -56,7 +56,7 @@ const AdherenceCalendar = observer(({ store }) => (
           date={date}
           medication_report={
             view === 'month' &&
-              store.medication_reports.records.find(mr =>
+              store.registration.information.medication_reports.find(mr =>
```
Comment (thecalliopecrashed) [inline src/components/AdherenceCalendar.js]: New way to access data!
More reliable!
More appropriate for how we're using it in the UI!

**Pull request merged**